daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DarReaderTest.scala

```diff
@@ -30,103 +30,54 @@ class DarReaderTest extends WordSpec with Matchers with Inside with BazelRunfile
-              ((packageId7, archive7), LanguageMajorVersion.V1) ::
-              ((packageId8, archive8), LanguageMajorVersion.V1) ::
-              ((packageId9, archive9), LanguageMajorVersion.V1) ::
-              ((packageId10, archive10), LanguageMajorVersion.V1) ::
-              ((packageId11, archive11), LanguageMajorVersion.V1) ::
-              ((packageId12, archive12), LanguageMajorVersion.V1) ::
-              ((packageId13, archive13), LanguageMajorVersion.V1) ::
-              ((packageId14, archive14), LanguageMajorVersion.V1) ::
-              ((packageId15, archive15), LanguageMajorVersion.V1) ::
-              ((packageId16, archive16), LanguageMajorVersion.V1) ::
-              ((packageId17, archive17), LanguageMajorVersion.V1) ::
-              ((packageId18, archive18), LanguageMajorVersion.V1) ::
-              Nil)) =>
-        packageId1 shouldNot be('empty)
-        packageId2 shouldNot be('empty)
-        packageId3 shouldNot be('empty)
-        packageId4 shouldNot be('empty)
-        packageId5 shouldNot be('empty)
-        packageId6 shouldNot be('empty)
-        packageId7 shouldNot be('empty)
-        packageId8 shouldNot be('empty)
-        packageId9 shouldNot be('empty)
-        packageId10 shouldNot be('empty)
-        packageId11 shouldNot be('empty)
-        packageId12 shouldNot be('empty)
-        packageId13 shouldNot be('empty)
-        packageId14 shouldNot be('empty)
-        packageId15 shouldNot be('empty)
-        packageId16 shouldNot be('empty)
-        packageId17 shouldNot be('empty)
-        packageId18 shouldNot be('empty)
-        archive1.getDamlLf1.getModulesCount should be > 0
-        archive2.getDamlLf1.getModulesCount should be > 0
-        archive3.getDamlLf1.getModulesCount should be > 0
-        archive4.getDamlLf1.getModulesCount should be > 0
-        archive5.getDamlLf1.getModulesCount should be > 0
-        archive6.getDamlLf1.getModulesCount should be > 0
-        archive7.getDamlLf1.getModulesCount should be > 0
-        archive8.getDamlLf1.getModulesCount should be > 0
-        archive9.getDamlLf1.getModulesCount should be > 0
-        archive10.getDamlLf1.getModulesCount should be > 0
-        archive11.getDamlLf1.getModulesCount should be > 0
-        archive12.getDamlLf1.getModulesCount should be > 0
-        archive13.getDamlLf1.getModulesCount should be > 0
-        archive14.getDamlLf1.getModulesCount should be > 0
-        archive15.getDamlLf1.getModulesCount should be > 0
-        archive16.getDamlLf1.getModulesCount should be > 0
-        archive17.getDamlLf1.getModulesCount should be > 0
-        archive18.getDamlLf1.getModulesCount should be > 0
+      case Success(Dar(main @ ((_, mainArchive), LanguageMajorVersion.V1), dalfDeps)) =>
+        forAll(main :: dalfDeps) {
+          case ((packageId, archive), _) =>
+            packageId shouldNot be('empty)
+            archive.getDamlLf1.getModulesCount should be > 0
+        }
 
```

Comment:
Maybe it is not important, but the check for `dalfDeps` version number is gone `LanguageMajorVersion.V1`

Reply:
I would just use `com.digitalasset.daml.lf.archive.Dar#all` instead of `main :: dalfDeps` in `forAll` and add acheck for language major version. But it is up to you. I see that you need `main` below.

Reply:
thanks, I’ve added the check for the version number and switched to `all`.